Mathlib/GroupTheory/SpecificGroups/Alternating/KleinFour.lean

```diff
@@ -0,0 +1,324 @@
+        norm_num
+        rw [← hα4, ← hlcm, Nat.card_eq_fintype_card]
+        apply le_trans (le_card_support_of_mem_cycleType hi)
+        apply Finset.card_le_univ
+    · rw [Nat.one_le_iff_ne_zero]
+      rintro ⟨rfl⟩
+      rw [hlcm, pow_zero] at hi
+      apply Nat.lt_irrefl 1
+      exact one_lt_of_mem_cycleType hi
+
+theorem card_of_four (hα4 : Nat.card α = 4) :
+    Nat.card (alternatingGroup α) = 12 := by
+  have : Nontrivial α := by
+    rw [← Finite.one_lt_card_iff_nontrivial, hα4]
+    norm_num
+  apply mul_right_injective₀ (by norm_num : 2 ≠ 0)
+  dsimp
+  rw [Nat.card_eq_fintype_card, two_mul_card_alternatingGroup,
+    Fintype.card_perm, ← Nat.card_eq_fintype_card, hα4]
+  rfl
+
+theorem card_two_sylow_of_four (hα4 : Nat.card α = 4) (S : Sylow 2 (alternatingGroup α)) :
+    Nat.card S = 4 := by
+  rw [Sylow.card_eq_multiplicity, card_of_four hα4]
+  have : 12 = 2 ^ 2 * 3 := by norm_num
+  rw [this, Nat.factorization_mul_apply_of_coprime (by decide), Nat.factorization_pow,
+    Finsupp.coe_smul, Pi.smul_apply, smul_eq_mul, Nat.prime_two.factorization_self,
+    Nat.factorization_eq_zero_of_not_dvd (by decide)]
+  norm_num
+
+theorem two_sylow_carrier_of_four (hα4 : Nat.card α = 4) (S : Sylow 2 (alternatingGroup α)) :
+    S.carrier =
+      {g : alternatingGroup α |
+        (g : Perm α).cycleType = 0 ∨ (g : Perm α).cycleType = {2, 2}} := by
+  classical
+  apply Set.eq_of_subset_of_card_le
+  · -- inclusion S ⊆ (cycleType = 0 ∨ cycleType = { 2, 2 })
+    intro k hk
+    simp only [Set.mem_setOf_eq]
+    obtain ⟨n, hn⟩ := (IsPGroup.iff_orderOf.mp S.isPGroup') ⟨k, hk⟩
+    rw [← orderOf_submonoid ⟨k, hk⟩, Subgroup.coe_mk] at hn
+    apply mem_kleinFour_of_order_two_pow hα4 k.prop
+    rw [orderOf_submonoid, hn]
+  -- card K4 ≤ card S
+  simp only [← Nat.card_eq_fintype_card]
+  change _ ≤ Nat.card S
+  rw [card_two_sylow_of_four hα4 S]
+  rw [Nat.card_eq_fintype_card]
+  apply le_trans (Fintype.card_subtype_or _ _)
+  simp only [Fintype.card_subtype]
+  simp only [AlternatingGroup.card_of_cycleType, ← Nat.card_eq_fintype_card, hα4]
+  decide
+
+theorem two_sylow_eq_kleinFour_of_four (hα4 : Nat.card α = 4) (S : Sylow 2 (alternatingGroup α)) :
+    S = kleinFour α := by
+  classical
+  apply le_antisymm
+  · intro k
+    rw [← Subgroup.mem_carrier, two_sylow_carrier_of_four  hα4 S, Set.mem_setOf_eq, or_imp]
+    constructor
+    · intro hk
+      suffices k = 1 by
+        rw [this]; exact Subgroup.one_mem _
+      rw [← Subtype.coe_inj, Subgroup.coe_one, ← cycleType_eq_zero, hk]
+    · intro hk
+      simp only [kleinFour]
+      apply Subgroup.subset_closure
+      simp only [Set.mem_setOf_eq]
+      exact hk
+  · simp only [kleinFour, Subgroup.closure_le]
+    intro g hg
+    simp only [SetLike.mem_coe, ← Subgroup.mem_carrier, two_sylow_carrier_of_four hα4 S]
+    apply Or.intro_right
+    simpa using hg
+
+theorem card_two_sylow_eq_one (hα4 : Nat.card α = 4) :
+    Nat.card (Sylow 2 (alternatingGroup α)) = 1 := by
```

Comment:
This is a bit of an odd spelling. Why not simply `Unique (Sylow 2 (alternatingGroup α))`?